### PR TITLE
Added AKS Platform Logs Rule

### LINKED ISSUE #922: Collect platform logs for AKS clusters
# Rule request

## Suggested rule change

Enable capture of platform diagnostics logs.

[Platform monitoring](https://docs.microsoft.com/azure/architecture/framework/devops/monitoring#platform-monitoring)
[Collect resource logs](https://docs.microsoft.com/azure/aks/monitor-aks#collect-resource-logs)

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.ContainerService/managedClusters**

Related to #882

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fix #922

Added `Azure.AKS.PlatformLogs` rule for the following types:

- Microsoft.ContainerService/managedClusters

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1 [hunk 1]
```diff
@@ -245,6 +245,83 @@ Rule 'Azure.AKS.AuditLogs' -Type 'Microsoft.ContainerService/managedClusters' -T
...
+            $_.category -in $availableLogCategories.Logs
+        });
+
+        $metricLogs = @($setting.Properties.metrics | Where-Object {
+            $_.enabled -and 
+            $_.category -in $configurationMetricCategories -and
+            $_.category -in $availableLogCategories.Metrics
+        });
+
+        $platformLogsEnabled = ($null -ne $platformLogs -and $platformLogs.Length -eq $logCategoriesNeeded) -and
```
Comment (ArmaanMcleod) [inline src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1:305]: I didn't check but can I use an Assertion here? Wasn't sure which one I can use.

Comment (BernieWhite) [reply src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1:305]: Technically you could use `$Assert.HasFieldValue` but like this `$Assert.HasFieldValue($platformLogs, 'Length', $logCategoriesNeeded)` or `$Assert.HasFieldValue($platformLogs.Length, '.', $logCategoriesNeeded)`.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice work!

I feel like we should be able to do this better in the future. I'll look at how we can extend PSRule language to make this less complex and verbose.

File: src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1 (hunk 1 above)
Comment (ArmaanMcleod) [reply src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1]: Thanks @BernieWhite. I've updated to use `HasFieldValue`.

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod Nice work!
> 
> I feel like we should be able to do this better in the future. I'll look at how we can extend PSRule language to make this less complex and verbose.

Thanks @BernieWhite. I've used your recommendation for `HasFieldValue` and looks quite good 🙂.

I think the assertions are great, I am getting a better understanding of how they work from just reading the source code. I think it could be a benefit to extend PSRule language to make it easier, but to be honest everything seems possible with them once you use them alot and read the source code.

**Pull request merged**